### ISSUE TITLE
Installation of FSL and ANTs

### ISSUE THREAD

**Issue by Fahadsal** [ubccr/software-layer#179]
Can someone please install FSL and ANTs software to /project/academic/schweser group? 

We specifically require the following versions,

FSL - 6.0.3
ANTs - 2.3.1

**Comment by Fahadsal** [comment]
I have tried installing myself but it happens to fail either way, using --robot to install with dependencies and even without (directly)

**Comment by dsajdak** [comment]
Hi @Fahadsal in the future, please put a separate request in for each software package as we're trying to share the load among our staff.  We will not install software into your group's project space.  We will attempt to install software with existing Easybuild recipes into CCR's software repository.  

If you want to try this again yourself, you will need to copy the existing Easybuild recipes into your directory, change the toolchain versions to match what CCR has to offer, rename the EB recipe to match that toolchain version, and possibly update the package dependencies within the recipe.  This way you can use what we already have and not rebuild toolchains or dependencies, wasting space in your project directory.  We have a list of currently supported toolchains [here](https://docs.ccr.buffalo.edu/en/latest/software/releases/).

**Comment by Fahadsal** [comment]
Sorry regarding the same request. Will make sure of it next time. As for the installation, thank you so much. Within CCR works, does not have to be in the project group, completely forgot it is more accessible through overall CCR. Thanks once again!  Regards,Fahad SalmanPhD Candidate Biomedical EngineeringResearch Assistant - Buffalo Neuroimaging Analysis CenterOn Jan 17, 2024, at 1:41 PM, Dori Sajdak ***@***.***> wrote:﻿
Hi @Fahadsal in the future, please put a separate request in for each software package as we're trying to share the load among our staff.  We will not install software into your group's project space.  We will attempt to install software with existing Easybuild recipes into CCR's software repository.
If you want to try this again yourself, you will need to copy the existing Easybuild recipes into your directory, change the toolchain versions to match what CCR has to offer, rename the EB recipe to match that toolchain version, and possibly update the package dependencies within the recipe.  This way you can use what we already have and not rebuild toolchains or dependencies, wasting space in your project directory.  We have a list of currently supported toolchains here.

**Comment by tonykew** [comment]
I have completed a test build of FSL - more info to follow.
My attempts to build ANTs have been, thus far, unsuccessful -still working on this...

**Comment by Fahadsal** [comment]
Thank you so much for your time and efforts. Really appreciate it!Regards,Fahad SalmanPhD Candidate Biomedical EngineeringResearch Assistant - Buffalo Neuroimaging Analysis CenterOn Jan 26, 2024, at 4:29 PM, Tony Kew ***@***.***> wrote:﻿
I have completed a test build of FSL - more info to follow.
My attempts to build ANTs have been, thus far, unsuccessful -still working on this...

**Comment by tonykew** [comment]
I have also completed a test build of ANTs - I had to skip the example
WarpVTKPolyDataMultiTransform.cxx  which would not compile.
More info to follow.

**Comment by tonykew** [comment]
FSL version 6.0.5.1 has been published to the ccrsoft/2023.01 software release

Note that this is a GUI app so run from CCR OnDemand

    https://ondemand.ccr.buffalo.edu

e.g.
[UB-HPC Cluster Desktop]
Partition [General Compute]
Quality of Service [General Compute]
[...]
Node Features "NOTLEGACY"


```
$ module spider FSL
----------------------------------------------------------------------------
  fsl: fsl/6.0.5.1
----------------------------------------------------------------------------
    Description:
      FSL is a comprehensive library of analysis tools for FMRI, MRI and
      DTI brain imaging data.

    Properties:
      Bioinformatic libraries/apps

    You will need to load all module(s) on any one of the lines below before the "fsl/6.0.5.1" module is available to load.

      gcc/11.2.0  openmpi/4.1.1
[...]
$ module load gcc/11.2.0 openmpi/4.1.1 fsl/6.0.5.1
$ fsl &
$ 

```

**Comment by tonykew** [comment]
ANTs version 2.5.1 has been published to the ccrsoft/2023.01 software release


```
vortex1$ module spider ants
----------------------------------------------------------------------------
  ants: ants/2.5.1
----------------------------------------------------------------------------
    Description:
      ANTs extracts information from complex datasets that include imaging.
      ANTs is useful for managing, interpreting and visualizing
      multidimensional data.

    Properties:
      Visualisation software

    You will need to load all module(s) on any one of the lines below before the "ants/2.5.1" module is available to load.

      gcc/11.2.0  openmpi/4.1.1
[...]
vortex1$ module load gcc/11.2.0 openmpi/4.1.1 ants/2.5.1
vortex1$ 

```


There are many binaries in the ANTs module:

```
vortex1$ ls ${EBROOTANTS}/bin
AddNoiseToImage
ANTS
antsAffineInitializer
antsAI
antsAlignOrigin
antsApplyTransforms
antsApplyTransformsToPoints
antsASLProcessing.sh
antsAtroposN4.sh
antsBOLDNetworkAnalysis.R
antsBrainExtraction.sh
antsCorticalThickness.sh
ANTSIntegrateVectorField
ANTSIntegrateVelocityField
antsIntermodalityIntrasubject.sh
antsIntroduction.sh
ANTSJacobian
antsJointFusion
antsJointLabelFusion.sh
antsJointTensorFusion
antsLandmarkBasedTransformInitializer
antsLaplacianBoundaryCondition.R
antsLongitudinalCorticalThickness.sh
antsMotionCorr
antsMotionCorrDiffusionDirection
antsMotionCorrStats
antsMultivariateTemplateConstruction2.sh
antsMultivariateTemplateConstruction.sh
antsNetworkAnalysis.R
antsNeuroimagingBattery
ANTSpexec.sh
antsRegistration
antsRegistrationSyNQuick.sh
antsRegistrationSyN.sh
antsSliceRegularizedRegistration
antsSurf
antsTransformInfo
ANTSUseDeformationFieldToGetAffineTransform
ANTSUseLandmarkImagesToGetAffineTransform
ANTSUseLandmarkImagesToGetBSplineDisplacementField
antsUtilitiesTesting
antsVol
Atropos
AverageAffineTransform
AverageAffineTransformNoRigid
AverageImages
AverageTensorImages
CheckTopology
ClusterImageStatistics
compareTwoTransforms
ComposeMultiTransform
CompositeTransformUtil
ConvertImage
ConvertImagePixelType
ConvertInputImagePixelTypeToFloat
ConvertScalarImageToRGB
ConvertToJpg
ConvertTransformFile
CopyImageHeaderInformation
CreateDisplacementField
CreateDTICohort
CreateImage
CreateJacobianDeterminantImage
CreateTiledMosaic
CreateWarpedGridImage
DenoiseImage
DeNrrd
ExtractRegionFromImage
ExtractRegionFromImageByMask
ExtractSliceFromImage
FitBSplineToPoints
GetConnectedComponentsFeatureImages
GetMeshAndTopology
ImageCompare
ImageIntensityStatistics
ImageMath
ImageSetStatistics
iMath
KellyKapowski
KellySlater
LabelClustersUniquely
LabelGeometryMeasures
LabelOverlapMeasures
LaplacianThickness
LesionFilling
MeasureImageSimilarity
MeasureMinMaxMean
MemoryTest
MultiplyImages
N3BiasFieldCorrection
N4BiasFieldCorrection
NonLocalSuperResolution
PasteImageIntoImage
PermuteFlipImageOrientationAxes
PrintHeader
RebaseTensorImage
ReorientTensorImage
ResampleImage
ResampleImageBySpacing
ResetDirection
sccan
SetDirectionByMatrix
SetOrigin
SetSpacing
simpleSynRegistration
SimulateDisplacementField
SmoothDisplacementField
SmoothImage
StackSlices
SuperResolution
SurfaceBasedSmoothing
SurfaceCurvature
TextureCooccurrenceFeatures
TextureRunLengthFeatures
ThresholdImage
TileImages
TimeSCCAN
waitForPBSQJobs.pl
waitForSGEQJobs.pl
waitForSlurmJobs.pl
waitForXGridJobs.pl
WarpImageMultiTransform
WarpTensorImageMultiTransform
WarpTimeSeriesImageMultiTransform
vortex1$ 

```

e.g.

```
vortex1$ antsRegistrationSyNQuick.sh -h

Usage:

antsRegistrationSyNQuick.sh -d ImageDimension -f FixedImage -m MovingImage -o OutputPrefix
[...]
vortex1$ 

```
